### DISCUSSION TITLE
The license just changed! Does it mean we can't run it in production anymore?

### DISCUSSION THREAD

**Discussion by mohaa7** [fingerprintjs/fingerprintjs#934] [General]
We would appreciate it if you could share your reasons for abandoning a GPL-like license.
We are a start-up (MVP stage) that _FingerprintJS_ is the core of our work! What should we do now?

**Comment by Valve** [comment]
Hi @mohaa7 
You can still use version 3.X.X in production, because it's still licensed under MIT.
However starting from version 4.0.0 we changed the license to BSL 1.1. that doesn't allow to run version 4 in production.
If you want to run version 4 in production, you need to purchase a commercial license.

> We would appreciate it if you could share your reasons for abandoning a GPL-like license.

I would like to clarify that we never licensed FingerprintJS under a GPL-like license. It was always licensed under MIT.
If you'd like to learn why we changed the license, please read our announcement here: https://fingerprint.com/blog/fingerprintjs-license-change/

**Comment by makma** [comment]
Hello @mohaa7,
All the reasons and implications of the license change are explained in our blog post at https://fingerprint.com/blog/fingerprintjs-license-change.

Cheers!

**Comment by mohaa7** [comment]
> You can still use version 3.X.X in production, because it's still licensed under MIT.

Yes, but my choice is definitely not a software with an exclusive license. I have to take the opinion of my team and check how to proceed. Thanks for your reply.

> we never licensed FingerprintJS under a GPL-like license. It was always licensed under MIT.

MIT is a non-copyleft free software license, compatible with the GNU GPL.

> All the reasons and implications of the license change are explained in our blog post at https://fingerprint.com/blog/fingerprintjs-license-change.

I see and respect your decision. I'm not going to flame, but meaningful to mention, you are no longer Open Source by definition as "[Open Source doesn’t just mean access to the source code](https://opensource.org/definition-annotated/)" and your blog post points this out in a subtle way by the sentence "keeping the source code available".

**Comment by princetechs** [comment]
hey @Valve , @mohaa7  Really it is the best lib in market, but am going tolive a opensource app, where i need it to give free credits to guest user, can i use it free in my opensource project ?

**Reply by JuroUhlar** [reply]
Hi @princetechs! The project's open-source nature does not matter as far as I understand. You can use Fingerprint for non-production, non-commercial, exploratory purposes but running it in production (as part of a revenue-generating project) requires a commercial license. Feel free to [contact our sales team](https://fingerprint.com/contact-sales/) to discuss if your project requires a commercial license.

**Comment by autronix** [comment]
I agree with the OP, fingerprintjs v4 does **NOT** qualify as open source anymore and shouldn't be presented as such on the README. To expand on the OP's link, section 6 of the [Open Source Definition](https://opensource.org/definition-annotated) clearly states:
```markdown
6. No Discrimination Against Fields of Endeavor
The license must not restrict anyone from making use of the program in a 
specific field of endeavor. For example, it may not restrict the program from 
being used in a business, or from being used for genetic research.
```

By introducing the commercial use licensing requirement, this violates the requirement to qualify for calling the project open source. 

In this case this repo should have stopped at v3 and forked into a new commercial license repo for v4 that doesn't pretend to be open source.

Broprint, available at https://github.com/Rajesh-Royal/Broprint.js, seems to provide similar functionality under MIT license without the commercial restrictions, as an actual Open Source project.

**Reply by JuroUhlar** [reply]
Hi @autronix, 

> I agree with the OP, fingerprintjs v4 does NOT qualify as open source anymore and shouldn't be presented as such on the README.

We agree. FingerprintJS v4 does not claim to be open-source, the README clearly states it's published under BSL and outlines which use cases require a license and which do not.

The only mention of "open-source" in the README is in reference to the open-source community. I have created a PR to remove that adjective to avoid any potential confusion: https://github.com/fingerprintjs/fingerprintjs/pull/1006 

Thank you!

**Comment by SyntaxDreamer** [comment]
The software and releases here on GitHub are under MIT license:
https://github.com/fingerprintjs/fingerprintjs/blob/master/LICENSE

And its even published as MIT on the GitHub repo.

**Comment by Valve** [comment]
We're in the process of updating the license from BSL to MIT in v5. We need to update all the links, docs etc (it's a lengthy process).
Please stay tuned and when you see an announcement in our blog (fingerprint.com/blog) you can start using v5 with all the updated docs under a permissive MIT license. Thanks
